Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -13,6 +13,95 @@
 
 namespace AZ
 {
+    namespace Internal
+    {
+        template <AZ::RHI::Format>
+        float RetrieveFloatValue(const AZ::u8* mem, size_t index)
+        {
+            AZ_Assert(false, "Unsupported pixel format");
+        }
+
+        template <AZ::RHI::Format>
+        AZ::u32 RetrieveUintValue(const AZ::u8* mem, size_t index)
+        {
+            AZ_Assert(false, "Unsupported pixel format");
+        }
```

Comment:
We should statically assert for unsupported formats.

Reply:
Will update to `static_assert`, thanks!